src/screens/home/swap-form/swap-form.tsx

```diff
@@ -68,6 +70,12 @@ export const SwapScreen = () => {
         outputAssetAddress
     );
 
+    useEffect(() => {
+        setInputAssetAmount(
+            formatNumber(Number(inputAssetAmount), inputAsset.decimals)
+        );
+    }, [inputAsset.decimals, inputAssetAmount, setInputAssetAmount]);
+
```

Comment:
it's bad behavior when `setValue` is called when `value` is changed.  This potentially could leak to cycled rerenders.  
just fix the value when it could be changed by the user